src/League/ApplicationConfigServiceProvider.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace TomPHP\ConfigServiceProvider\League;
+
+use League\Container\ServiceProvider\AbstractServiceProvider;
+use TomPHP\ConfigServiceProvider\ApplicationConfig;
+use TomPHP\ConfigServiceProvider\ApplicationConfigIterator;
+
+final class ApplicationConfigServiceProvider extends AbstractServiceProvider
+{
+    /**
+     * @var ApplicationConfig
+     */
+    private $config;
+
+    /**
+     * @var string
+     */
+    private $prefix;
+
+    /**
+     * @param ApplicationConfig $config
+     * @param string            $prefix
+     */
+    public function __construct(ApplicationConfig $config, $prefix)
+    {
+        $this->prefix   = $prefix;
+        $this->config   = $config;
+        $this->provides = array_map(
+            function ($key) {
+                return $this->keyPrefix() . $key;
+            },
+            $config->getKeys()
+        );
+    }
+
+    public function register()
+    {
+        $container = $this->getContainer();
+        $iterator  = new ApplicationConfigIterator($this->config);
+        $prefix    = $this->keyPrefix();
+
+        foreach ($iterator as $key => $value) {
+            $this->getContainer()->share($prefix . $key, function () use ($value) {
```

Comment:
@tomphp 

Since we extend from `League\Container\ServiceProvider\AbstractServiceProvider`, we could just use the instance property `$container`, what do you think?

Alternatively, we could use the temporary variable `$container`, which you have introduced above.

Either way, we could save a bunch of method calls here. 😄